packages/flutter_tools/lib/src/device.dart

```diff
@@ -993,3 +997,12 @@ class NoOpDevicePortForwarder implements DevicePortForwarder {
   @override
   Future<void> dispose() async { }
 }
+
+String computeDartVmFlags(DebuggingOptions debuggingOptions) {
+  String flags = debuggingOptions.dartFlags ?? '';
+  if (debuggingOptions.nullAssertions) {
+    final String separator = flags.isEmpty ? '' : ',';
+    flags += '$separator--null_assertions';
```

Comment:
Is making a list and `join`ing too clunky?

Reply:
Very difficult for my sleepy brain 🤣

Reply:
I just wanted you to know that I understand the fence post problem

Reply:
fixed